src/EditorFeatures/VisualBasicTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.vb

```diff
@@ -0,0 +1,429 @@
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)>
+        Public Async Function TestBinaryExpression_TestAvailableWithAlwaysRemove_And_NotAvailableWhenRequiredForClarity2() As Task
+            Await TestAsync(
+"class C
+    sub M()
+        dim q = c + $$(a * b)
+    end sub
+end class",
+"class C
+    sub M()
+        dim q = c + a * b
+    end sub
+end class", requireAllParenthesesForClarity:=False)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)>
+        Public Async Function TestForOverloadedOperatorOnLeft() As Task
+            Await TestInRegularAndScript1Async(
+"class C
+    sub M(c1 as C, c2 as C, c3 as C)
+        dim x = $$(c1 + c2) + c3
+    end sub
+
+    public shared operator +(c1 as C, c2 as C) as C
+        return nothing
+    end operator
+end class",
+"class C
+    sub M(c1 as C, c2 as C, c3 as C)
+        dim x = c1 + c2 + c3
+    end sub
+
+    public shared operator +(c1 as C, c2 as C) as C
+        return nothing
+    end operator
+end class", parameters:=New TestParameters(options:=RequireAllParenthesesForClarity))
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)>
+        Public Async Function TestMissingForOverloadedOperatorOnRight() As Task
+            Await TestMissingAsync(
+"class C
+    sub M(c1 as C, c2 as C, c3 as C)
+        dim x = c1 + $$(c2 + c3)
+    end sub
+
+    public shared operator +(c1 as C, c2 as C) as C
+        return nothing
+    end operator
+end class", parameters:=New TestParameters(options:=RequireAllParenthesesForClarity))
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)>
+        Public Async Function TestShiftRequiredForClarity1() As Task
+            Await TestMissingAsync(
+"class C
+    sub M()
+        dim x =  $$(1 + 2) << 3
+    end sub
+end class", parameters:=New TestParameters(options:=RequireShiftParenthesesForClarity))
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)>
+        Public Async Function TestShiftRequiredForClarity2() As Task
+            Await TestMissingAsync(
+"class C
+    sub M()
+        dim x = $$(1 + 2) << 3
+    end sub
+end class", parameters:=New TestParameters(options:=RequireAllParenthesesForClarity))
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)>
+        Public Async Function TestRemoveShiftIfNotNecessary1() As Task
+            Await TestInRegularAndScript1Async(
+"class C
+    sub M()
+        dim x = $$(1 + 2) << 3
+    end sub
+end class",
+"class C
+    sub M()
+        dim x = 1 + 2 << 3
+    end sub
+end class", parameters:=New TestParameters(options:=RemoveAllUnnecessaryParentheses))
+        End Function
+    End Class
+End Namespace
```

Comment:
Test idea: `Dim x = $$(y = z)` and conversely `Dim x = y $$= z`

Reply:
Added. thanks.  #resolved